Angular/Angular guidelines and best practices/File and module organization and naming.md

```diff
@@ -15,53 +15,113 @@ Try to keep the code grouped by domain. For example, if you have a `Post` class,
+- `/posts` - list of posts (renders a list of `post-overview` and `user-avatar` components)
+- `/posts/new` - new post form (renders `post-form`)
+- `/posts/:id/edit` - edit post form (renders `post-form`)
+- `/posts/:id/view` - post details view (renders `post-overview`, `user-avatar` and `post-content` components)
 
-- Post component is used in `posts-container` (list of posts) and `users-container` (user's list of posts) top-level routes:
+Directory structure for such an app should look like this:
 
 ```
 src/app/
-├── components/
-│   └── post/
-└── pages/
-    ├── users-container/
-    └── posts-container/
-```
-
-- `PostsCount` component is used only inside the `users-container` top-level route, but it is used in both `user-details-container` and `users-list-container` sub-routes:
-
-```
-src/app/
-└── pages/
-    └── users-container/
-        ├── components/
-        │   └── posts-count/
-        ├── user-details-container/
-        └── users-list-container/
+├── components/ # these components are used in both posts and users feature modules
+│   ├── post-overview/ # renders basic info about a post (without full content)
+│   └── user-avatar/ # renders username and profile image
+├── services/ # these services are used in both posts and users feature modules
+│   ├── auth/ # session management, not important for this example
+│   ├── user/
+│   │   ├── user.service.ts # has methods for data fetching
+│   │   ├── user.service.spec.ts
+│   │   ├── user.testing.service.ts
+│   │   ├── user.interface.ts
+│   │   └── user.model.ts
+│   └── post/ # basically the same as user.service
+├── pages/
+│   ├── users/
+│   │   ├── pages/
+│   │   │   ├── users-index/
+│   │   │   │   ├── components/
+│   │   │   │   │   └── users-list/ # renders a list of user-avatar components
+│   │   │   │   ├── users-index.component.ts # renders some header/footer and users-list component
+│   │   │   │   ├── users-index-routing.module.ts # eagerly-loads users-index.component on /
+│   │   │   │   └── users-index.module.ts # imports users-index-routing.module
+│   │   │   └── user-index/
+│   │   │       ├── components/
+│   │   │       │   └── user-details/ # renders user-avatar component, additional user info and a list of user's posts (post-overview)
+│   │   │       ├── user-index.component.ts # renders some header/footer and users-details component
+│   │   │       ├── user-index-routing.module.ts # eagerly-loads user-index.component on /
+│   │   │       └── user-index.module.ts # imports user-index-routing.module
+│   │   ├── users-routing.module.ts # lazy-loads users-list.module on /, user-details.module on /:id/view
+│   │   └── users.module.ts # imports users-routing.module
+│   └── posts/
+│       ├── components/ # these components are used only in posts feature/domain module
+│       │   ├── post-form/ # form used for both editing and creation
+│       │   └── post-content/ # renders post content
```

Comment:
I now noticed that `post-content` is used only in post-details component so it doesn't really need to be this high up in the folder hierarchy. I will think about moving it deeper in the hierarchy or adding another place where it will be used

Reply:
> > There is one point of discussion around naming, regarding `-index` components. Previoulsy we would probably name these `-container`, We need _some_ postfix because otherwise there are components with duplicate names. For example, we need a naming distinction between `user-details(1)` component that is loaded via routing (eagerly) and `user-details(2)` component that actually renders user details. `user-details(1)` component renders some sort of a header/footer, maybe fetches some data from the API and renders `user-details(2)` with that data. After a discussion with @kristijanbambir and taking a look at some other projects, `-index` seems to be a good option. Thusly, we named `user-details(1)` `user-index` and `user-details(2)` is simply `user-details`. I would like to hear some other opinions.
> 
> I'm used to `-container` naming but other than that I don't have any objections to the `-index` naming. @fvoska You mentioned that some projects are using the `-index` convention, could you list some of those projects (I'm just curious)?

[Polyglot](https://github.com/infinum/js-polyglot-angular/tree/master/src/app/pages/projects/pages) used component name index, although it was not as a postfix but as the complete name.